signer/spiffe/spiffe.go

```diff
@@ -45,3 +54,96 @@ func Signer(ctx context.Context, socketPath string) (cryptoutil.Signer, error) {
 
 	return cryptoutil.NewSigner(svid.PrivateKey, cryptoutil.SignWithIntermediates(svid.Certificates[1:]), cryptoutil.SignWithCertificate(svid.Certificates[0]))
 }
+
+func DelgatedSigner(ctx context.Context, socketPath string, workloadPath string) (cryptoutil.Signer, error) {
+	client, err := getDelgatedClient(ctx, socketPath)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("#####: %s", socketPath)
+
+	binPath, err := exec.LookPath(workloadPath)
+	if err != nil {
+		return nil, err
+	}
+
+	hasher := sha256.New()
+	s, err := ioutil.ReadFile(binPath)
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	_, err = hasher.Write(s)
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	h := hex.EncodeToString(hasher.Sum(nil))
+
+	selector := types.Selector{
+		Type:  "unix",
+		Value: fmt.Sprintf("sha256:%s", h),
+	}
+
+	selectors := []*types.Selector{
+		&selector,
+	}
+
+	req := &delegatedidentityv1.SubscribeToX509SVIDsRequest{
+		Selectors: selectors,
+	}
+
+	stream, err := client.SubscribeToX509SVIDs(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := stream.Recv()
+	if err != nil {
+		return nil, err
+	}
+
+	svids := resp.X509Svids
```

Comment:
svids isn't used outside of the line below this, so this can just be `svid := resp.X509Svids[0]`.